odoo/addons/test_new_api/tests/test_properties.py

```diff
@@ -133,6 +133,24 @@ def test_properties_field(self):
         self.assertEqual(self.message_3.read(['attributes'])[0]['attributes'], expected)
         self.assertEqual(self.message_3.attributes, expected)
 
+        # check that the keys not valid for the given type are removed
```

Comment:
Note that we could make a dedicated unit test (better have small unit tests than bit tests), as you prefer (and as framework team prefers). Like ``test_properties_field_parameters_cleanup`` for example.

Reply:
better indeed :)